src/FSharp.AWS.DynamoDB/TableContext.fs

```diff
@@ -1402,6 +1318,141 @@ type TableContext<'TRecord>
+            ?precondition: ConditionExpression<'TRecord>
+        ) : Transaction =
+        setClient tableContext.Client
+        let req = Put(TableName = tableContext.TableName, Item = tableContext.Template.ToAttributeValues item)
+        precondition
+        |> Option.iter (fun cond ->
+            let writer = AttributeWriter(req.ExpressionAttributeNames, req.ExpressionAttributeValues)
+            req.ConditionExpression <- cond.Conditional.Write writer)
+        transactionItems.Add(TransactWriteItem(Put = req))
+        this
+
+    /// <summary>
+    ///   Adds a ConditionCheck operation to the transaction.
+    /// </summary>
+    /// <param name="tableContext">Table context to operate on.</param>
+    /// <param name="key">Key of item to check.</param>
+    /// <param name="condition">Condition to check.</param>
+    member this.Check(tableContext: TableContext<'TRecord>, key: TableKey, condition: ConditionExpression<'TRecord>) : Transaction =
+        setClient tableContext.Client
+
+        let req = ConditionCheck(TableName = tableContext.TableName, Key = tableContext.Template.ToAttributeValues key)
+        let writer = AttributeWriter(req.ExpressionAttributeNames, req.ExpressionAttributeValues)
+        req.ConditionExpression <- condition.Conditional.Write writer
+        transactionItems.Add(TransactWriteItem(ConditionCheck = req))
+        this
+
+    /// <summary>
+    ///   Adds an Update operation to the transaction.
+    /// </summary>
+    /// <param name="tableContext">Table context to operate on.</param>
+    /// <param name="key">Key of item to update.</param>
+    /// <param name="updater">Update expression.</param>
+    /// <param name="precondition">Optional precondition expression.</param>
+    member this.Update
+        (
+            tableContext: TableContext<'TRecord>,
+            key: TableKey,
+            updater: UpdateExpression<'TRecord>,
+            ?precondition: ConditionExpression<'TRecord>
+
+        ) : Transaction =
+        setClient tableContext.Client
+
+        let req = Update(TableName = tableContext.TableName, Key = tableContext.Template.ToAttributeValues key)
+        let writer = AttributeWriter(req.ExpressionAttributeNames, req.ExpressionAttributeValues)
+        req.UpdateExpression <- updater.UpdateOps.Write(writer)
+        precondition |> Option.iter (fun cond -> req.ConditionExpression <- cond.Conditional.Write writer)
+        transactionItems.Add(TransactWriteItem(Update = req))
+        this
+
+    /// <summary>
+    ///  Adds a Delete operation to the transaction.
+    /// </summary>
+    /// <param name="tableContext">Table context to operate on.</param>
+    /// <param name="key">Key of item to delete.</param>
+    /// <param name="precondition">Optional precondition expression.</param>
+    member this.Delete
+        (
+            tableContext: TableContext<'TRecord>,
+            key: TableKey,
+            precondition: option<ConditionExpression<'TRecord>>
```

Comment:
better to make this be `?precondition: ConditionExpression<'TRecord>` and add an OptionalAttribute as that means
a) F# users can omit the argument
b) C# users can do the same

Same for the other args